### DISCUSSION TITLE
Possible balance fixes and feature additions

### DISCUSSION THREAD

**Discussion by induktio** [induktio/thinker#32] [Ideas]
This list contains various ideas that attempt to address imbalances in the game mechanics or add new features to the game. Simple bugs that contradict the original game documentation or clearly skip the intended game mechanics belong to the other thread "original game engine bugs".

* Stack collateral damage reduction from defaults. Additionally probe teams currently have 100% collateral damage and this cannot be changed from the config. This should be adjustable using a separate config option.
* Fix mineral contribution exploits/unbalanced mechanics when disbanding units, artifacts or hurrying units. These costs should not change if INDUSTRY rating is changed multiple times per turn.
* Make AI unable to see cloaked units and subs (pressure hulls).
* Alter the model/mesh to designate between “fighter” needlejets (with SAM ability) and “bomber” needlejets (without SAM ability).
* Improve automatic worker tile priority. Sometimes (usually through F4->Citizens->set doctor to worker) worker was set to 1-food, 1-energy tile when a 3-food, 1-energy tile was available.
* The battle odds could be better explained. Perhaps the percentages of either side winning the battle with a certain HP remaining?
* Alter unit movement after dropping to a certain HP. Add more fine-grained options on how damage affects unit speed.
* Show an unmet faction’s borders before meeting them, even if the border is some generic color as to not give away which faction it is. 
* Have a “go back to the END of previous turn” save game. This is to avoid going back to fix silly things you forgot before clicking next turn.
* Show currently used/maximum possible amounts for police and free unit support on base screen. Implementation might be complicated.
* Show current clean mineral count visually either on the base screen or somewhere on the HQ/base operations menu.
* Optional feature to have ships count as police units.
* Add option to disable tech holes (1/3 chance of not being able to research some tech). This should not be enabled by default since it might interfere with the AIs if too many of them start researching the same techs.
* Ensure that interceptor/ship prototypes are created properly for each faction. Sometimes they might be missing (more details?)
* Add air superiority mission support for AI interceptors.
* Possibly rewrite decision making for AI planet buster attacks.
* Add logic to parse default AI faction strategy settings "emphasize land/sea/air/lots of probe team etc".

Changes only in alphax.txt:

* Secret project rebalancing. Increase costs for important projects such as Planetary Transit System/Weather Paradigm or move them to the later techs.

Miscellaneous bugs:

* [UNVERIFIED] Recently recreation commons and hologram theaters just straight up disappear from large bases. There were no enemy probe teams, drone riots, or energy shortages. Is it possible for drones to destroy buildings on the same turn the riot happens?
* [UNVERIFIED] Some battles, particularly with needlejets and probe teams, don’t even give a warning pop up—your unit just straight up dies and it’s avoidable and incredibly frustrating.

**Comment by Michael2589** [comment]
A feature that I think would be useful would be a generic mod folder parameter.  You already have the basic functionality in the SMAC-only option. This would just be a parameter such as 'mod_folder="my mod".  This would be useful to players in general, and in particular you could put the sorts of game balance changes discussed in this thread into a mod folder and players could easily choose to use the mod.

A supplementary feature could include the ability to set the mod's title in a file in the mod folder, and have that mod title show up as part of the report when the player presses Alt-T.

Now I have experimented with rebalancing secret projects, with combinations of increasing costs and moving the projects to later techs. What I concluded, after a few games, was that, at least for single player games, increasing the costs of secret projects just slowed the pace of the game down, in a bad way, and that the SP's are generally in appropriate places in the tech tree. There were obviously exceptions to this, so for example I moved the Universal Translator down to a tier 3 tech and I think that assigning it to a tier 1 or 2 tech could easily be justified, and I moved the Cloudbase Academy to a tier 13 tech because I think it removes from the strategy of the game. (Aerospace complexes are one of the best base facilities, but they are expensive, so deciding where to build them first is an interesting strategic decision, and just having them everywhere for free makes that trivial, besides being OP.)  And the Planetary Transit System should be moved to a different tech (not necessarily a higher tier) than the one that's a prerequisite for crawlers.

Now if you are designing a mod specifically for very experienced players using directed research, then large scale changes to SP tech requirements and costs could certainly be justified, but I don't think that would be a good change for a mod aimed at a general audience.

In the rules that I use for my own games I have settled on giving supply modules a cost (in alphax terms) of 20, which means that unarmoured fission crawlers cost 60 and unarmoured fusion crawlers cost 30. This works well for me because it makes crawlers available but rare between the discovery of industrial automation and the discovery of fusion power, and after fusion power.  I still build crawlers, but I'm certainly not using them just to crawl forests for 2 minerals/turn, which is definitely viable when crawlers cost 30.  (This doesn't work so well for AI's using Thinker mod, because they still spend as many or more resources building 60-mineral crawlers as they would building 30-mineral crawlers, and then when they get Fusion Power they armor their crawlers because that's good design in the vanilla game, but things like this are why we give the AI cheats.)

However, my first choice for nerfing crawlers would be to make them cost maintenance like most other units.

Ah, needlejets and copters. I think the biggest problem with needlejet balance is that the SAM ability has the same prereq tech as the needlejet chassis. SAM rovers trade extremely well against fission impact penetrators. For that matter, fission impact penetrators don't do well at all against defensive infantry with equivalent tech, and they cost as much maintenance and upset the drones as much as better aircraft do, so I consider them to be a niche unit. It's just that if you're leading in tech, and you've just researched needlejets, then no one else can attack them.

I guess you could justify a small increase in needlejet chassis costs, but I don't see it as a major issue.  Maybe make the prerequisite for SAM Synthetic Fossil Fuels, or even Doctrine: Flexibility [edit] or Applied Physics or Nonlinear Mathematics, because those techs are prerequisites for new weapons so it could thematically make sense for some of those new weapons to be useable as AA weapons[/edit].

Copters are a necessary game mechanic to counter the tactic of using numerous cheap units to exploit the fact that most unit types can only attack one target per turn, no matter how much a top-of-the-line unit overpowers one of the numerous cheap units. Copters are a bit overpowered, but they have a hard counter in the form of defensive infantry (which is one of the reasons I nag you so much when defensive infantry leave their bases!), and if a copter's going to use its ability to retreat after combat then it can't attack far from its home base.

When I was experimenting with rebalancing copters, I reduced their base speed to 6 compared to needlejets' 8 (so fusion copters had 8 compared to fusion needlejets' 10, etc.), and give the copter chassis the same cost as a needlejet chassis, although of course making them a bit more expensive than needlejets works too.

The inconsistency issue with needlejet combat mechanics sounds like a bug, and could go in the "original game engine bugs" thread.

The addition to Thinker's map generator sounds like a nice idea for people playing multi-player.

**Reply by induktio** [reply]
Yeah, so the above list references issues that might be in theory be implemented in the future, but it's by no means guaranteed. In some form or another, the mod will probably always have the default tech tree available as an option. Also most people maybe play with the default tech tree for nostalgia reasons, so I'm not sure how many players even prefer those kind of changes. So far the mod has done only minor edits to alphax.txt variables.

About the "generic mod folder parameter", I'm not sure if there's a sufficiently clear use case to warrant implementing that. It might just make the usability even more confusing if additional game config files are added.

There's also another idea to nerf the copters, instead of reducing the movement points, we could rewrite the rules such that Copters expend double movement points when they attack something. That would somewhat nerf the attack capability and not require reducing the range excessively compared to needlejets.

**Comment by ghost** [comment]
Out of your list, I'd love to see fractional movement point mag tubes. I still keep them disabled because they just make it too easy to blitz an AI once I can get on their network, but I miss the convenience of slightly faster transportation.

By the sound of it, diplomacy changes sound complicated but what I would like to see is the AI not offer to trade a tech for your maps. It's just a free tech for the player while the AI gets nothing of actual value to itself. I'm pretty sure I've also been offered this trade by a pact mate, too, which is doubly useless for the poor AI.

**Comment by induktio** [comment]
Looks like in the next build we'll have the option to reactivate units that were skipped on the same turn. In order to avoid issues, it should be only possible reactivate units that did not move at all when they were skipped. Otherwise it would require using an extra struct field to save the fractional movement units which is not a very good use for the space. Overall this feature can be done but it requires a minor change in the savegame format.

**Comment by induktio** [comment]
There are a couple of features that could use more research. Normally Thinker's AI checks for unit visibility flag maintained by the game engine whenever it chooses attack targets. This flag should be the same that determines if the unit on a tile is visible to another faction if the fog of war is on. If the unit is inside a base, it is treated as always visible, but otherwise visibility rules should apply.

Particularly for air units, this affects whether the AI will beeline to attack other units that are more far away. There have been some reports that Deep Pressure Hull and Cloaking Device don't affect AI vision at all. It would be useful to know if anyone has tested these abilities with Thinker AI in more detail. Are there any specific situations when these abilities don't seem to have effect on the AI or is it always the same?

**Comment by PFCWilliamLHudson** [comment]
Thank you for your excellent mod!
Please, consider adding these fixes/features:
1) An option to disable diplomatic comm breaks from sunspot events. All other effects from the event are ok, but inability to negotiate sometimes becomes very inconvenient.
2) Solution for chop&drop tactic. An option that attacks from air units against land units can't reduce defender's health below 10% outdoor, 50% in a base or bunker. Probe teams and non-combat units (formers, crawlers) aren't subject to restriction and can be killed.
3) An option for probe teams to not exert ZoC, and to not provide circumvention of the enemy ZoC. Also separate option that does the same for air units.

**Comment by Rothe123** [comment]
I absolutely love this mod.. there is one feature that I have always wanted especially have civ4 came out to disable "TECH TRADING" at first I thought maybe just add a sunspots permanently but then that would not allow even talking to anyone.. so if you could just add an option in thinker.ini to disable just tech trading that would to me make this a perfect mod.. and thanks for your work!

**Reply by induktio** [reply]
Looks like there's already an option "No technology trading" in Scenario Rules (Shift+F11). This should probably do what you described, but it seems it requires starting a new scenario that has this setting enabled. There's no option to enable it for random map starts.

**Comment by MacNokker** [comment]
Something I've always found strange about hasty attacks is that rovers don't seem to be able to retreat from other rovers conducting them even though they could retreat from a damaged rover attacking with its "full" 1MP.

The behaviour should be consistent in that rovers can either never retreat from other mobile types or perhaps be based completely on the attackers current MP instead. 

Another small improvement would be to remove the "hasty attack" and "odds against us) warnings when confirm attacks is enabled, or make an option to remove them.

There`s also this weird interaction between Planetary Transit System's "no drones in bases <= pop 3" that completely negates FM's pacifism drones. So with a little micro your airforce can be as big as you like.

**Reply by induktio** [reply]
About the rovers only sometimes being able to retreat from other rovers, should this be caused by the other rover being damaged and having reduced movement? In that case it should count as being an unit of slower speed, so don't see an issue with this behavior. Retreat is only possible when this condition is true (plus some others that are not listed here)

   veh_speed(veh_id_def, 1) > veh_speed(veh_id_atk, 1)

The movement points being deducted after enough damage should be a clearly intended and documented game mechanic. Hasty attack dialog could be possibly hidden based on some options since this is easy to modify with the engine rewrites.

Planetary Transit System should reduce drones by one on bases of size three or smaller. It does not remove all drones nor prevent pacifism or bureaucracy drones from appearing. This part of code should be checked so there's unlikely to be issues.

**Reply by MacNokker** [reply]
Yes it does work for damaged rovers (max MP 1), but it does not affect hasty attacks by rovers.

So you can retreat from a normal attack by infantry (1/1 MP) but not from a hasty attack by a rover (0.x/2 MP).
In that case the rover has already used its mobility advantage to even engage in battle at all. It should not have the ability to prevent retreat because it technically has negative MP.

About the PTS, you are right, I thought it worked like that but I just read the UI wrong.

**Reply by induktio** [reply]
Well, not sure if the hasty attack retreat mechanic should be classified as an actual bug. This could be modded but the feature is somewhat borderline when it comes to specific bugfixes unless this contradicts some known documentation. Are there references on this on the manual? In this case if the intention was to check for full movement points then it could be left as it is. Just saying generally this project tries to make a distinction between simple bugfixes and modifications and this affects which changes should have their own config options.

**Reply by MacNokker** [reply]
The name of this topic is "Possible balance fixes and feature additions". :)

While it's not a bug, it is inconsistent behaviour, since you can retreat from a damaged rover.

**Reply by induktio** [reply]
It's mostly a question of which changes to prioritize and which ones to implement as options but this would be one of the smaller changes. Anyway this can be considered but not sure yet if it would be implemented.

**Reply by Michael2589** [reply]
This is actually described in the manual (page 98 under "Disengagement") and it is working as described in the manual:

"In combats where the defending unit is faster than the attacker (i.e., the defender has at least one more movement point than the other), the faster unit can disengage if the battle is going against it..."

Since a sufficiently badly damaged rover would start the battle with a speed of 1, a defending rover with a speed of 2 would be able to disengage from it (if it meets the other requirements to qualify for disengaging: there are some) the same as it would be able to disengage from other units with a speed of 1.

**Comment by JediMindTrix** [comment]
my big suggestion would be allowing to queue movements/actions by holding down something like ctrl+shift while picking from a dropdown list. it would make it easier on the mind to keep a consistent plan going for base development, exploration, avoiding fungus, etc
I think it'd be a massive QoL feature. Even just being able to queue three commands max

**Comment by JediMindTrix** [comment]
Another suggestion would be to add a shortcut that when held down that lets you plot a road for formers to build, instead of having to do it a tile at a time

**Reply by induktio** [reply]
There is already Construct Road To (Ctrl+R) former command that should do just that. It should also work with modded formers unless there is some issue.

**Reply by induktio** [reply]
Regarding the other point the game has patrols for units with up to four waypoints. The unit will keep repeating the loop so it's not just one time route though. Rewriting this would probably need reimplementing the whole unit movement code on the user interface which is also related to things mentioned on issue https://github.com/induktio/thinker/issues/107 .

**Reply by JediMindTrix** [reply]
Okay no problem.
In regards to the Construct Road To, I was hoping to add a visual indicator to it basically so I *know* how it'll plan to lay the road out
